lib/brainstem/query_strategies/base_strategy.rb

```diff
@@ -42,14 +40,35 @@ def calculate_per_page
 
       private
 
-      def get_ids_sql(scope)
+      def ids_and_count_for_page(page, scope, count_scope, should_paginate)
+        if use_calc_row? && @options[:paginator]
+          scope = primary_presenter.apply_ordering_to_scope(scope, @options[:params])
+          ids, count = @options[:paginator]&.get_paged(page, scope)
```

Comment:
`@options[:paginator]&` The safe operator might not be needed since you verified `@options[:paginator]` exists in the conditional.